10/RequestReply.md

```diff
@@ -77,6 +77,11 @@ SHOULD include:
 #### Exception
 
 #### Timeout
+
+* On the client side, the Request-Reply protocol SHALL implement timeout mechanism for every type of request. Timeout mechanism is returning DevFailed expetion if a request cannot be served in specified time called Timeout.
+* For each request the client MAY setup Timeout value or opt-out from using timeout mechanism which is equal to set Timeout to infinity.
```

Comment:
(At least) in the C++ version, setting a timeout to 0 disables it.
1 thing important to know (in C++ at least) is that timeout involves only the requests going through the network.
If a Device Server is a client of a device exported by the same device server (same process), there are some optimizations in omniORB which is calling the methods directly without going through the network.
In this case, if the method takes more time than the configured timeout, no timeout exception will be raised and the client will be blocked until the end of the method execution.

Reply:
Regarding omniORB implementation for local call, is its behaviour (blocking) an expected one or should it be marked as kind of bug?

Reply:
I will rephrase the sentence with infinity to avoid confusions in respects to how APIs define opt-out fro, timeout.

Reply:
> 
> 
> Regarding omniORB implementation for local call, is its behaviour (blocking) an expected one or should it be marked as kind of bug?

It is expected but not well known I think. So I thought this is a good opportunity to mention this potentially misleading behaviour.
I'm actually not sure this behaviour is well documented on readthedocs.